### ISSUE TITLE
Feature List: Second Release

### ISSUE THREAD

**Issue by thadius856** [RaspberryPints/RaspberryPints#174]
Before commenting here, please provide feedback on the last release (hotwash) in #173.

What features do we want to bring to the table for v2?

Here's what is currently listed in our milestone for Second Release:
- [ ] **Flow meters:** Database edits to support flow meters volume recording and calibration.
- [ ] **Flow meters:** Support pulse counting and recording in SQL. (_kerber_)
- [ ] **Flow meters:** Provide calibration method in admin panel. (_kerber_)
- [ ] **Documentation:** Instructions for flow meter wiring
- [ ] **Documentation:** Instructions for breakout board assembly.
- [ ] **Documentation:** Instructions for flow meter calibration
- [ ] **Frontend:** Provide live updates on changes to SQL db using jQuery and AJAX.
- [ ] **Customization:** Provide ability to turn all frontend data/images on/off individually using admin panel. (_thadius856_)
- [ ] **Customization:** Provide _preliminary_ support for skinning for later release. (_skemp45_)
- [ ] **Customization:** Allow for custom drink colors, other than those available in SRM.
- [ ] **BeerXML:** Support import of BeerXML files to `beers` database.
- [ ] ~~**Postponed:** Support for RFID reader and drinker accounts.~~ postponed to v3.0.0

What should stay? What should go? What is missing? If you're wanting to contribute a feature, please specify that you want to code it.

I'll close in 72 hours, at which point we can set the timeframe for development.

**Comment by Boombrewer** [comment]
Here are my suggestions (notice I said suggestions since I do absolutely no coding)...
- Update instructions on how to install update without losing data
- Thad's recipe fields on the beer page (they're empty now, so let people fill it in)

That's really all I can think of for now.  @thadius856 - What are you thinking for the V2 release date?  You guys have some big goals up there.

**Comment by thadius856** [comment]
Well, I start a mandatory 5-week leadership course on Monday, with lots of extra-curriculars, so I'm rather out of the picture for the next 5 weeks or so.

**Comment by Skemp45** [comment]
YEs, I need a slight break from the project just to soak it all in. lets get a solid list of stuff we want for v2 and then decide as a group how long it will take us to deliver

**Comment by Skemp45** [comment]
After looking at the list I think drinker accounts should wait for v3, We should really be focusing on getting the flow meter support and the ability to update current setups without loosing data. This will ensure we release in a timely manner and get all hands on deck for a few things and make sure they work really well.

**Comment by Skemp45** [comment]
I do agree with a few other small feature additions.

**Comment by thadius856** [comment]
Good. Scratched drinker accounts to v3.

Adding more customization options for v2. I think personalized layouts will be important, and I'd like to get a jump on skinning. Besides, there's not a whole lot of stuff up above that I'm capable of completing.

**Comment by Skemp45** [comment]
What kind of skinning are you looking for? like admin screen colors or fonts???

**Comment by thadius856** [comment]
Fonts, sizes, colors, spacing, length. Yep.

I don't plan to finish it by v2 release, but I'd like to start steps towards making it possible for our community to build and share their own skins.

**Comment by kerber** [comment]
This sounds like a really good focus for V2. I've been making a lot of progress on the Arduino side. Lots of testing and lots of forward thinking.

**Comment by Skemp45** [comment]
humm, I like your thoughts there @thadius856 allow users to make themes and share them, like WordPress..... HUMMMM let me look into theme support. and @kerber nice work man, keep at it

**Comment by thadius856** [comment]
Updated above. Please take a look and let me know if I missed any projects.

**Comment by Skemp45** [comment]
Looks perfect

**Comment by Skemp45** [comment]
we need to makesure we release a changelog.txt with our code releases

**Comment by thadius856** [comment]
I was going to append it to `README.md`, but yeah, we should be doing this.

Anybody else have projects they want to work on this release? It's ok to take a break if not.

**Comment by mtnbound** [comment]
We can add the beer.xml as well since it's fairly close to done.

Just a thought, but rather than locking V2 down this soon, why don't we just add a priority to the features. Since it's only been released a few days, it would be interesting to see what the community wants for features going forward. We can let the community know the priority and they can always comment on the list.

Might even be worth asking in HBT if anyone else wants to contribute as well. Since you have the branching structure as feature-xxx anyway, if something doesn't get done, or the code is poor, there's no need to merge it back. 

I saw someone mention they were an mobile developer. That might be very helpful if we add "drinker accounts", or "Bottle Lists" since the user could tap on a picture, use NFC, or scan a bottle label, etc. etc. Probably not V2 for sure, but might be good for the list later.

**Comment by Skemp45** [comment]
I am cool with having a few more devs but we need to be careful on having too many hands in the cookie jar. we have a good flow and understanding of each other now and it works well. lets not spoil that

**Comment by Skemp45** [comment]
And it is good to lock down what you want in a version otherwise things can get too sidetracked. its best to know when something needs to be done and work towards it, always have an end goal so you can have checkpoints

**Comment by mtnbound** [comment]
Completely agree. I was just thinking if we add the branches off the main develop branch with "feature-xxx" it would keep it separated well, and protect against poor code.

Might also be worth adding a QA/QC step before adding any branch back into the master.

At work, we never let developers merge their own code :)

**Comment by Skemp45** [comment]
^ Smart work lol

**Comment by thadius856** [comment]
I was pondering this idea, but guess I never posted it.

@Skemp45, is there some existing WP plugin that would let anons submit feature requests, and then vote on which ones they care about?

It'd be really nice to see a community-developed, stratified list of wants (even if we disagree and don't implement).

**Comment by mtnbound** [comment]
Can we maybe create a Wiki page here on Git for a feature dump. I think each of us have had ideas come to mind. Might be a good place for all of us to be able to add them. Maybe a section for actual "approved" features, and then a section for "thoughts" or drunken ramblings. That way there's a place to dump all the ideas each of us have.

**Comment by thadius856** [comment]
How about... this?

http://www.wpmayor.com/plugin-reviews/best-feature-request-and-vote-plugin-for-wordpress/

**Comment by Skemp45** [comment]
Yeah, worst case we can take the suggestions and make a poll and have them vote on it

**Comment by thadius856** [comment]
```
One of the best ways of enabling users to suggest features is to use a system whereby
a user can submit his idea, and other users can vote it up or down. Thus the most
highly voted ideas or feature requests are prime candidates for future development on
your theme/plugin.
```

Yeahhhhh, I'd really appreciate crowdsourcing feature requests. :D

**Comment by jkurl** [comment]
your feature list is spot on.  For me one of the biggest features I am looking forward to is the flow meters. untapped is coming along and the themes is moving along nicely.  someone mentioned a mobile developer but I have viewed this on my Nexus 7 and other devices and it looks nice even on them.  the flow meters might be a issue for the mobile devices and not update properly.  

   I am not real big on a lot of frills, so I like what you guys currently have for features and asking the masses will help get buy in for this project and give people a sense of I helped create this project(not sure these are the right words). you already have a lot of buy in but the project potential is huge there is no ceiling right now for this taplist.
